SpatialGDK/Source/SpatialGDKFunctionalTests/Private/GenerateTestMapsCommandlet.cpp

```diff
@@ -0,0 +1,276 @@
+// everything after themselves, so other tests can run with a clean slate.
+void UGenerateTestMapsCommandlet::CreateSpatialNetworkingMap()
+{
+	ULevel* CurrentLevel = CreateNewTestMapWorld()->GetCurrentLevel();
+
+	// Add the tests
+	GEditor->AddActor(CurrentLevel, ASpatialTestPossession::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ASpatialTestRepossession::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ASpatialTestRepNotify::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, AVisibilityTest::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARPCInInterfaceTest::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARegisterAutoDestroyActorsTestPart1::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ARegisterAutoDestroyActorsTestPart2::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, AOwnerOnlyPropertyReplication::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ADormancyAndTombstoneTest::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, ASpatialTestSingleServerDynamicComponents::StaticClass(), FTransform::Identity);
+
+	// Add test helpers
+	// Unfortunately, the nature of some tests requires them to have actors placed in the level, to trigger some Unreal behavior
+	GEditor->AddActor(CurrentLevel, ADormancyTestActor::StaticClass(), FTransform::Identity);
+	GEditor->AddActor(CurrentLevel, AReplicatedVisibilityTestActor::StaticClass(), FTransform::Identity);
+
+	FString PathToSave = FPaths::ConvertRelativePathToFull(FPaths::ProjectContentDir()
+														   + TEXT("Maps/FunctionalTests/GeneratedMaps/CI_Fast/SpatialNetworkingMap.umap"));
+	FEditorFileUtils::SaveLevel(CurrentLevel, PathToSave);
+}
+
+// This map is a simple 2-server-worker map, where both workers see everything in the world
+void UGenerateTestMapsCommandlet::CreateSpatial2WorkerMap()
+{
+	UWorld* CurrentWorld = CreateNewTestMapWorld();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	// Add the tests
+
+	// The two orchestration tests are placed at opposite points around the origin to "guarantee" they will land on different workers so
+	// that they can demonstrate they work in all situations
+	GEditor->AddActor(CurrentLevel, ACrossServerAndClientOrchestrationTest::StaticClass(), FTransform(FVector(-250, 250, 0)));
+	GEditor->AddActor(CurrentLevel, ACrossServerAndClientOrchestrationTest::StaticClass(), FTransform(FVector(250, -250, 0)));
+	// This one lives in the TestGyms, TODO
+	// GEditor->AddActor(CurrentLevel, APredictedGameplayCuesTest::StaticClass(), FTransform::Identity);
+	// This one lives in the TestGyms, TODO
+	// GEditor->AddActor(CurrentLevel, ACrossServerAbilityActivationTest::StaticClass(), FTransform::Identity);
```

Comment:
@KeyMaster- these are your tests

Reply:
These tests have 2 requirements on the project that's hosting them:
1. The code module they're in should list GAS as a dependency. I believe if it doesn't but the project still uses GAS anyway, they actually compile fine (though with a warning), but I expect that if the project doesn't use GAS otherwise the compile would fail.
2. The project has certain tags listed in its DefaultGameplayTags.ini file. If it doesn't, schema gen will fail on an assert when certain objects request those tags.

What project these tests live in is irrelevant beyond that. So maybe we can move these tests into the project that now runs the functional tests?

Reply:
We could, but there's another question further down in this PR, which is whether to even use *that* project. We could turn to using the starter template for example.

Reply:
Mhh, that's probably worth figuring out first then. If it's a project which could host tests that can't be in the GDK, putting these in that repo would make sense. If we land on something like the starter template which can't just include random tests, we could consider storingt the code etc in another repo (e.g. the GDK) without actually including them in any compile/run step, and copying them from the GDK into the project during CI.